src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -90,7 +90,7 @@ module sonic-device_metadata {
                 leaf type {
                     type string {
                         length 1..255;
-                        pattern "ToRRouter|LeafRouter|SpineChassisFrontendRouter|ChassisBackendRouter|ASIC";
+                        pattern "ToRRouter|LeafRouter|SpineChassisFrontendRouter|ChassisBackendRouter|ASIC|MgmtToRRouter|SpineRouter|BackEndToRRouter|BackEndLeafRouter|EPMS|MgmtTsToR";
```

Comment:
since pattern is limited, is length limitation necessary?

Reply:
not sure if there was some discussion/reasoning about the length limit when this yang model was created. Hence prefer to leave it as is.